### PR TITLE
Added new functionality to be able to cancel a search and to set a maximum number of subfolders to search in

### PR DESCRIPTION
**Pull request by ericf-axosoft**
## Summary of changes
- Added `maxSubfolderDeep` parameter.
- Modified the callback to be able to return true to cancel the search.
- Fix compiling node module with spaces in path.
- Updated readme file.
- Added new tests.
- Fixed `GitHub` action for the tests.

### CODE REVIEW

**Comment by ericf-axosoft** [comment]
@ianhattendorf could you please take a look at this PR? 
I'm not able to add reviewers.

Thanks!

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     name: Tests
     strategy:
         matrix:
-          node: [10, 12]
+          node: [18, 20, 22]
           os: [windows-2016, ubuntu-16.04, ubuntu-18.04, macOS-latest]
```
Comment (ianhattendorf) [inline .github/workflows/tests.yaml]: You'll need to update the OSes as well, the Windows and Linux OSes here aren't available on GitHub Actions any more.

https://github.com/actions/runner-images/tree/main/images/ubuntu
https://github.com/actions/runner-images/tree/main/images/windows

**Comment by ericf-axosoft** [comment]
@ianhattendorf I also don't know why only one of the test nodes has been executed. The others are waiting all the time.

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (ericf-axosoft) [reply .github/workflows/tests.yaml]: Done

**Review by ianhattendorf — APPROVED** [review]

**Pull request merged**